### PR TITLE
KFSPTS-29211 Fix ad hoc routed docs going to EXCEPTION

### PR DESCRIPTION
**Pull request by cah292**
This is ready for initial review, but I would like to do a little more local testing before it gets merged.

### CODE REVIEW

**Comment by cah292** [comment]
I've finished my testing efforts locally and in Sandbox for now; further validation can be performed in KFS DEV. This is now ready to be merged once it has been reviewed and approved.

File: src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java
Comment (jhulslander) [inline src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java:34]: I don't understand why we want to throw this exception.  If the method hasn't been overriden, won't that break routing

File: src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml
Comment (jhulslander) [inline src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml:127]: When I look for message.delivery in kuali-puppet, I only see it in some bash SH files.  Is this setting needed elsewhere

**Review by jhulslander — COMMENTED** [review]
I had a couple of questions

File: src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java
Comment (cah292) [reply src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java:34]: I was trying to find a way to avoid breaking other potential compilation problems while still indicating that it would be best to explicitly implement this method. However, I agree that the thrown exception is confusing, so I'll update this method to default to true.

File: src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml:127]: In the financials project, the default KFS config file sets `message.delivery` to asynchronous mode, whereas the default integration test config file sets it to synchronous mode. These defaults should be appropriate and we don't need to override them elsewhere.

If you notice a spot where a non-unit-test environment is trying to set up synchronous processing instead, please let me know.

Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml:127]: Take a look at https://github.com/search?q=repo%3ACU-CommunityApps%2Fkuali-puppet%20message.delivery&type=code .... looks like batch_job_up.sh is used, we should double check that

File: src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java
Comment (jhulslander) [inline src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java:34]: maybe we should log this event as you mentioned it shouldn't happen and is a sign of something awry.  We can set up an alert in datadog if the log entry is found

File: src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml
Comment (jhulslander) [reply src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml:127]: looking at the code in https://github.com/CU-CommunityApps/kuali-puppet/blob/5bbd11f427df3d607ad9fd2f421c5de046edc016/files/batch_scripts/batch_job_up.sh#L22 I don't think it would ever set  mode="-Dmessage.delivery=synchronous"

Comment (cah292) [reply src/main/resources/edu/cornell/kfs/kew/config/cu-spring-kew.xml:127]: I agree that I'm not seeing any setup that would result in the `-Dmessage.delivery=synchronous` option getting chosen, unless I missed something. Also, I confirmed in the KFS DEV logs that this morning's eInvoice Route Job was performing the `StandardWorkflowEngine` work in a different thread than the one that routed the document. It seems like we already have asynchronous processing set up in the KFS batch system.

File: src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java
Comment (cah292) [reply src/main/java/org/kuali/kfs/kew/api/document/DocumentProcessingQueue.java:34]: The problem with adding a log statement here is that this is an interface, not a class. As a result, a static `LOG` constant in this spot would have to be declared as public; Java doesn't let interfaces have private static constants.

If this default method somehow got invoked, I think the worst case is that the FINP-8541 feature won't be effective anymore. (That feature isn't a high priority; it just adds a Complete request to a SAVED document if such a request doesn't exist for some reason.) I doubt this default method would ever get invoked, and even if it did, it seems like the consequences of its execution would not be serious. Thus, we might not need to add logging here.

Thoughts?

**Comment by cah292** [comment]
I committed a change to have the default method simply return true instead of throwing an exception. Please also see my other comments, where I agree that we don't need to worry about synchronous processing somehow being enabled, and where we might not need to add logging in that default method.

@jhulslander If you agree that no further changes are necessary, then this PR can be merged once it has been approved.

**Review by jhulslander — APPROVED** [review]
my concerns have been addressed, I shall merge

**Pull request merged**
